pkg/cli/admin/upgrade/status/health.go

```diff
@@ -0,0 +1,144 @@
+package status
+
+import (
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+)
+
+type scopeType string
+
+const (
+	scopeControlPlane scopeType = "ControlPlane"
+	scopeWorkerPool   scopeType = "WorkerPool"
+)
+
+type allowedScopeKind string
+
+const (
+	clusterOperator allowedScopeKind = "ClusterOperator"
+)
+
+type scopeResource struct {
+	kind      allowedScopeKind
+	namespace string
+	name      string
+}
+
+type updateInsightScope struct {
+	scopeType scopeType
+	resources []scopeResource
+}
+
+type impactLevel uint32
+
+const (
+	infoImpactLevel impactLevel = iota
+	warningImpactLevel
+	errorImpactLevel
+	criticalInfoLevel
+)
+
+func (l impactLevel) String() string {
+	switch l {
+	case infoImpactLevel:
+		return "Info"
+	case warningImpactLevel:
+		return "Warning"
+	case errorImpactLevel:
+		return "Error"
+	case criticalInfoLevel:
+		return "Critical"
```

Comment:
What is the difference between error and critical? We also need go doc for each type. It is almost like an API because once we define these we can not change it significantly. Users will setup automation or monitoring system based on these.

Reply:
I basically followed common logging levels where you usually have one level above error. I imagine critical is something where the cluster or data is actually at risk of being destroyed or lost. I'm not married to this. We are not committed to the client-side "api" but I guess adding notes about what we, as implementors, intend to use these levels for is useful, will add.

Reply:
Added a commit with godocs